ts/components/services/ServicesTab.tsx

```diff
@@ -0,0 +1,271 @@
+  ) => void;
+  onItemSwitchValueChanged: (service: ServicePublic, value: boolean) => void;
+  tabOffset: Animated.Value;
+}>;
+
+type Props = OwnProps &
+  ReturnType<typeof mapStateToProps> &
+  ReturnType<typeof mapDispatchToProps> &
+  LightModalContextInterface;
+
+const styles = StyleSheet.create({
+  activeTextStyle: {
+    ...makeFontStyleObject(Platform.select, "600"),
+    fontSize: Platform.OS === "android" ? 16 : undefined,
+    fontWeight: Platform.OS === "android" ? "normal" : "bold"
+  },
+  textStyle: {
+    fontSize: customVariables.fontSizeSmall
+  },
+  organizationLogo: {
+    marginBottom: 0
+  },
+  icon: {
+    paddingHorizontal: (24 - 17) / 2 // (io-right icon width) - (io-trash icon width)
+  }
+});
+
+function renderOrganizationLogo(organizationFiscalCode: string) {
+  return (
+    <OrganizationLogo
+      logoUri={getLogoForOrganization(organizationFiscalCode)}
+      imageStyle={styles.organizationLogo}
+    />
+  );
+}
+
+function organizationContainsText(item: Organization, searchText: string) {
+  return isTextIncludedCaseInsensitive(item.name, searchText);
+}
+
+class ServicesTab extends React.PureComponent<Props> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {};
+    this.onSaveAreasOfInterest = this.onSaveAreasOfInterest.bind(this);
+    this.showChooserAreasOfInterestModal = this.showChooserAreasOfInterestModal.bind(
+      this
+    );
+    this.renderLocalQuickSectionDeletion = this.renderLocalQuickSectionDeletion.bind(
+      this
+    );
+  }
+  /**
+   * For tab Locals
+   */
+  private showChooserAreasOfInterestModal() {
+    const {
+      selectableOrganizations,
+      hideModal,
+      selectedOrganizations
+    } = this.props;
+
+    const OrganizationsList = createFactory(
+      ChooserListContainer<Organization>()
+    );
+    this.props.showModal(
+      <OrganizationsList
+        items={selectableOrganizations}
+        initialSelectedItemIds={some(new Set(selectedOrganizations || []))}
+        keyExtractor={(item: Organization) => item.fiscalCode}
+        itemTitleExtractor={(item: Organization) => item.name}
+        itemIconComponent={left(renderOrganizationLogo)}
+        onCancel={hideModal}
+        onSave={this.onSaveAreasOfInterest}
+        isRefreshEnabled={false}
+        matchingTextPredicate={organizationContainsText}
+        noSearchResultsSourceIcon={require("../../../img/services/icon-no-places.png")}
+        noSearchResultsSubtitle={I18n.t("services.areasOfInterest.searchEmpty")}
+      />
+    );
+  }
+
+  private onSaveAreasOfInterest(selectedFiscalCodes: Option<Set<string>>) {
+    if (this.props.updateOrganizationsOfInterestMetadata) {
+      if (this.props.updateToast) {
+        this.props.updateToast();
+      }
+      this.props.updateOrganizationsOfInterestMetadata(selectedFiscalCodes);
+    }
+    this.props.hideModal();
+  }
+
+  private renderLocalQuickSectionDeletion(section: ServicesSectionState) {
+    const onPressItem = () => {
```

Comment:
This _const_ doesn't solve the _arrow function in a render context_: **onPressItem** will be created each time _renderLocalQuickSectionDeletion_ is fired (and it is fired in render method). Furthermore declaring a function inside another function decreases the code readability
So this case seems to be the case when we can't avoid arrow function. But for sure we can keep an high level of code readability.
So my suggest is:

- take out **onPressItem** and make it an instance method
```typescript
private onPressItem = (section: ServicesSectionState) => {
    if (this.props.userMetadata && this.props.selectedOrganizations) {
      if (this.props.updateToast) {
        this.props.updateToast();
      }
      const updatedAreasOfInterest = this.props.selectedOrganizations.filter(
        item => item !== section.organizationFiscalCode
      );
      this.props.saveSelectedOrganizationItems(
        this.props.userMetadata,
        updatedAreasOfInterest
      );
    }
  };
```
- then call it using an arrow function
```typescript
<TouchableOpacity onPress={() => this.onPressItem(section)}>
        <IconFont
          name={"io-trash"}
          color={customVariables.brandMildGray}
          size={17}
          style={styles.icon}
        />
      </TouchableOpacity>
```

Reply:
The committed code reverts to the usage of lambda functions, trying to avoid inputs being passed to the function. Moreover, a new constant is isolated (getting the height of the header depending on the os)